lib/task/import/validate/csvDigitalObjectPathValidator.class.php

```diff
@@ -0,0 +1,232 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV Digital Object Path and URI check. Check digitalObjectPath and report:
+ *  - images not referenced from CSV
+ *  - images referenced in CSV but not found in image folder
+ *  - images referenced more that once in the CSV.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvDigitalObjectPathValidator extends CsvBaseValidator
+{
+    const TITLE = 'Digital Object Path Test';
+
+    // Do not reset between CSVs.
+    protected $fileList = [];
+    protected $pathToDigitalObjects = '';
+
+    protected $digitalObjectPathColumnPresent;
+    protected $digitalObjectUriColumnPresent;
+    protected $digitalObjectUses = [];
+    protected $overriddenByUriCount = 0;
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+        parent::__construct($options);
+
+        $this->setPathToDigitalObjects($this->options['pathToDigitalObjects']);
+    }
+
+    public function reset()
+    {
+        $this->digitalObjectPathColumnPresent = null;
+        $this->digitalObjectUses = [];
+        $this->digitalObjectUriColumnPresent = null;
+        $this->overriddenByUriCount = 0;
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        if (!isset($this->digitalObjectPathColumnPresent)) {
+            $this->digitalObjectPathColumnPresent = array_key_exists('digitalObjectPath', $row);
+        }
+
+        if (!isset($this->digitalObjectUriColumnPresent)) {
+            $this->digitalObjectUriColumnPresent = array_key_exists('digitalObjectUri', $row);
+        }
+
+        if ($this->digitalObjectPathColumnPresent) {
+            if (!empty($row['digitalObjectPath'])) {
+                $this->addToUsageSummary($row['digitalObjectPath']);
+            }
+
+            // URI is preferred by import CLI task if both path and uri are populated.
+            if ($this->digitalObjectUriColumnPresent) {
+                if (!empty($row['digitalObjectPath']) && !empty($row['digitalObjectUri'])) {
+                    ++$this->overriddenByUriCount;
+                }
+            }
+        }
+    }
+
+    public function getTestResult()
+    {
+        if (false === $this->digitalObjectPathColumnPresent) {
```

Comment:
Is it not valid to have CSV file with a `digitalObjectUri` column, but no `digitalObjectPath` column?